### ISSUE TITLE
SIM103: false positive

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2399]
```
def foo(exc):
    if exc.getcode() == 401:
        return False
    else:
        return False
```
causes:
```foo.py:2:5: SIM103 Return the condition `exc.getcode() == 401` directly```

**Comment by charliermarsh** [comment]
Definitely a bug, but where did you run into this :joy:

**Comment by spaceone** [comment]
https://github.com/univention/univention-corporate-server/commit/37e560e5282fab764ff6496131b28bf0d492b900#diff-45af649ef64d42abac0d51f47809588eaf0aff8501ff6fc0dc6a8580134f25c1R141-R144 ;-)

**Comment by ngnpope** [comment]
Amazing 😆 

`SIM103` definitely doesn't make sense. It could become:

```python
def foo(exc):
    exc.getcode()  # Still have to call because of potential side effects.
    return False
```

But then there could also be side effects if someone overloaded `__eq__` to do some evil... 🙈 

Maybe a check to identify that all branches have the same statements within?
